### ISSUE TITLE
it will go into an endless loop when analyzing the fieldaccessExpr as a.a.a

### ISSUE THREAD

**Issue by zyj183247166** [javaparser/javaparser#2702]
Using JavaParser to resolve the type of `a.a.a(a fieldaccessExpr)` , it will go into an endless loop.
The analyzed program that has expressions as a.a.a is as follow:
```java
  public String toString() {
    return String.format(
        "Fp12 (%s; %s)\n" +
            "     (%s; %s)\n" +
            "     (%s; %s)\n" +
            "     (%s; %s)\n" +
            "     (%s; %s)\n" +
            "     (%s; %s)\n",

        a.a.a, a.a.b,
        a.b.a, a.b.b,
        a.c.a, a.c.b,
        b.a.a, b.a.b,
        b.b.a, b.b.b,
        b.c.a, b.c.b
    );
  }
```

You can use the `FieldAccessExpr` of JavaParser to analyze `a.a.a` and try to resolve the type of the first `a` or the second `a`. And you will find that it go into an endless loop.

**Comment by zyj183247166** [comment]
If you need the analyzed program and the program that is written using JavaParser, you can contact me. zyj183247166@qq.com
I will send you the program that is written using JavaParser and the analyzed program that has the FieldAccessExpr a.a.a.
You just run the program, and it will go into an endless loop.
I have debug the program and find the bug is in FieldAccessContext.java.
You just need debug the program and step by step you will find why the loop occurs.

**Comment by zyj183247166** [comment]
So how to correct this error?
Just not to analyze the type of a.a.a. So before analyze the type of some node or expression, add some statements to judge if it belongs to the type of a.a.a(FieldAccessExpr)
`
private ResolvedType getNodeDescribeRT(Node node) {
	        ResolvedType rt = null;
	        if(node instanceof FieldAccessExpr) {
	        	Expression scope =((FieldAccessExpr) node).getScope();
				if(scope instanceof FieldAccessExpr) {
					return null;
				}
	        }      
	        try {
	        	//System.out.println(node.toString());
	        	rt = JavaParserFacade.get(combinedTypeSolver).getType(node);
	        } 
	        //catch (RuntimeException e) {
	        catch (Exception e) {
	            // skip
	        }
	        return rt;
}
`

**Comment by MysterAitch** [comment]
Apologies, I tend to work out of GitHub notifications and somehow missed this one -- nothing personal in ignoring you, I promise! :smile:

I don't have an easy answer to hand, but I do seem to recall reading somewhere that symbol resolution:
1. removes the last element of the reference/name
2. tries to resolve it
3a. if it is resolved, great -- used it
3b. if it is NOT resolved, jump to (1) by removing the last element of the reference/name

Assuming that this is ACTUALLY the case, it seems reasonable to assume there's a bug here when every element within the package names is identical -- that's where I would be inclined to look. Or it might be that I'm talking rubbish! :+1: 

Bottom line: I don't have the answer, but I'll mark this as a bug.

**Comment by jlerbsc** [comment]
Hi zyj183247166 , Could you elaborate because the test case below

    @BeforeEach
    void setUp() {
        TypeSolver typeSolver = new ReflectionTypeSolver();
        ParserConfiguration config = new ParserConfiguration();
        config.setSymbolResolver(new JavaSymbolSolver(typeSolver));
        StaticJavaParser.setConfiguration(config);
    }

    @Test
    void test2702() {
        String code = 
                "class a {" +
                "  a a;" +
                "  a m1() {" +
                "    return a.a;" +
                "  }" +
                "}";
      
        CompilationUnit cu = StaticJavaParser.parse(code);
        FieldAccessExpr fae = cu.findFirst(FieldAccessExpr.class).get();
        ResolvedValueDeclaration resolved = fae.resolve();

        System.out.println(resolved.getType());
        
    }

returns

    ReferenceType{a, typeParametersMap=TypeParametersMap{nameToValue={}}}

**Comment by jlerbsc** [comment]
Hi @zyj183247166 could you tell us if you still have this problem. If yes, can you provide us with simple unit test.

**Comment by jlerbsc** [comment]
I'm closing this issue because we don't have a way to reproduce this problem. You are free to reopen it if you have new information to communicate to us.
